tests_tf/test_attacks.py

```diff
@@ -44,6 +57,175 @@ def test_generate_np(self):
         # test perturbation norm
         self.assertTrue(np.allclose(perturbation_norm, self.attack.eps))
 
+class TestFastGradientMethod(CleverHansTest):
+    def setUp(self):
+        super(TestFastGradientMethod, self).setUp()
+        import tensorflow as tf
+
+        # The world's simplest neural network
+        def my_model(x):
+            W1 = tf.constant([[1.5, .3], [-2, 0.3]], dtype=tf.float32)
+            h1 = tf.nn.sigmoid(tf.matmul(x, W1))
+            W2 = tf.constant([[-2.4, 1.2], [0.5, -2.3]], dtype=tf.float32)
+            res = tf.nn.softmax(tf.matmul(x, W2))
+            return res
+
+        self.sess = tf.Session()
+        self.model = my_model
+        self.attack = FastGradientMethod(self.model, sess=self.sess)
+
+    def test_generate_np_gives_adversarial_example(self):
+        x_val = np.random.rand(100, 2)
+        x_val = np.array(x_val, dtype=np.float32)
+
+        for ord in [np.inf, 1, 2]:
```

Comment:
This is the first test of its kind in this repo so it may be worth discussing how we want to approach "parameterized" tests here. I think you are right about not writing 3 tests w/ the exact same code for the 3 norms.

Would there be an alternative way to write this so the error is easier to understand once Travis fails on this test (in other words, so we can easily tell which norm caused the failure).